repository/orm.go

```diff
@@ -63,29 +64,27 @@ func (r *ormRepository) GetErrors(serviceName string, numberOfErrors int) ([]Err
 
 // ReplaceErrors deletes previous stored errors for a service name and stores the new list of errors in json format
 func (r *ormRepository) ReplaceErrors(serviceName string, errors []ErrorAggregate) {
-	err := r.DB.Transaction(func(tx *gorm.DB) error {
-		// Delete previous records
-		if err := tx.
+	for _, errorAggregate := range errors {
+		errObj := AggregatedError{}
+		key := errorAggregate.AggregationKey
+		result := r.DB.Model(&AggregatedError{}).
 			Where("service_name = ?", serviceName).
-			Unscoped().
-			Delete(&AggregatedError{}).Error; err != nil {
-			return err
+			Where("aggregation_key = ?", key).
+			First(&errObj)
+		if result.RowsAffected == 0 {
+			r.DB.Create(&AggregatedError{
+				ServiceName:    serviceName,
+				Errors:         errorAggregate,
+				AggregationKey: key,
+				TotalCount:     errorAggregate.TotalCount,
+			})
+		} else if errorAggregate.TotalCount > errObj.TotalCount { // only update if there are more errors than before
+			r.DB.Model(&AggregatedError{}).
+				Where("service_name = ?", serviceName).
+				Where("aggregation_key = ?", key).
+				Update("total_count", errorAggregate.TotalCount).
+				Update("Errors", errorAggregate)
```

Comment:
unrelated to the scope of the PR, but just noticed this now. the column name uses uppercase title casing while others use snake case. is this on purpose?

Reply:
Absolutely not and might be a bug, thanks!

Reply:
It wasn't, sql is usually case insetive. I changed it anyway